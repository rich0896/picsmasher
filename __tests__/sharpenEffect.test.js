/* eslint-disable no-undef */
// __tests__/sharpenEffect.test.js

const { SharpenEffect } = require('../effects.js'); // Adjust the path as needed
const { applyEffectAndGetImageData } = require('../helpers/testHelpers.js'); // Adjust path if different

describe('SharpenEffect', () => {
    let ctx, canvas;

    beforeEach(() => {
        // Create a canvas element using the global document provided by jsdom
        canvas = document.createElement('canvas');
        canvas.width = 10; // Reduced size for smaller snapshots
        canvas.height = 10;
        ctx = canvas.getContext('2d');

        // Fill the canvas with a solid color (e.g., rgb(100, 150, 200))
        ctx.fillStyle = 'rgb(100, 150, 200)';
        ctx.fillRect(0, 0, canvas.width, canvas.height);
    });

    test('should apply sharpen correctly at intensity 1', () => {
        const effect = new SharpenEffect({ intensity: 1 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should apply sharpen correctly at intensity 3', () => {
        const effect = new SharpenEffect({ intensity: 3 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });

    test('should handle zero intensity without changes', () => {
        const effect = new SharpenEffect({ intensity: 0 });
        const serializedImageData = applyEffectAndGetImageData(effect, ctx, canvas);

        expect(serializedImageData).toMatchSnapshot();
    });
});
