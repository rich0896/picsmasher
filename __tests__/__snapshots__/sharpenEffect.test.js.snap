// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharpenEffect should apply sharpen correctly at intensity 1 1`] = `ImageData(10, 10, [255, 255, 255, 255, 200, 255, 255, 255, 200, 255...])`;

exports[`SharpenEffect should apply sharpen correctly at intensity 3 1`] = `ImageData(10, 10, [255, 255, 255, 255, 255, 255, 255, 255, 255, 255...])`;

exports[`SharpenEffect should handle zero intensity without changes 1`] = `ImageData(10, 10, [255, 255, 255, 255, 200, 255, 255, 255, 200, 255...])`;
